pkg/controller/walle/controller.go

```diff
@@ -0,0 +1,443 @@
+	if exists {
+		obj.(*NodeInformer).close()
+		wr.nodeInformerMap.Delete(key)
+	}
+}
+
+func revertKey(key string) (string, string) {
+	s := strings.Split(key, "/")
+	if len(s) == 2 {
+		return s[0], s[1]
+	}
+	return "", ""
+}
+
+// GetStore returns the SharedIndexInformers cache.Store for a given kluster or an error
+func (wr *WalleController) GetStore(kluster *v1.Kluster) (cache.Store, error) {
+	informer, err := wr.getNodeInformerByKluster(kluster)
+	if err != nil {
+		return nil, err
+	}
+	return informer.GetStore(), nil
+}
+
+// GetIndexer returns the SharedIndexInformers cache.Indexer for a given kluster or an error
+func (wr *WalleController) GetIndexer(kluster *v1.Kluster) (cache.Indexer, error) {
+	informer, err := wr.getNodeInformerByKluster(kluster)
+	if err != nil {
+		return nil, err
+	}
+	return informer.GetIndexer(), nil
+}
+
+func (wr *WalleController) getNodeInformerByKluster(kluster *v1.Kluster) (NodeInformer, error) {
+	key, err := cache.MetaNamespaceKeyFunc(kluster)
+	if err != nil {
+		return NodeInformer{}, err
+	}
+	informer, ok := wr.nodeInformerMap.Load(key)
+	if ok {
+		return informer.(NodeInformer), nil
+	}
+	return NodeInformer{}, fmt.Errorf("no informer found for kluster %v", key)
+}
+
+// List..
+func (wr *WalleController) List() map[*v1.Kluster]cache.SharedIndexInformer {
+	m := make(map[*v1.Kluster]cache.SharedIndexInformer)
+	wr.nodeInformerMap.Range(
+		func(key, value interface{}) bool {
+			ns, name := revertKey(key.(string))
+			k, err := wr.Clients.Kubernikus.KubernikusV1().Klusters(ns).Get(name, meta_v1.GetOptions{})
+			if err != nil {
+				wr.logger.Log("Couldn't get kluster %s/%s while listing all", ns, name)
+			}
+			m[k] = value.(NodeInformer).SharedIndexInformer
+			return true
+		},
+	)
+	return m
+}
+
+func (wr *WalleController) HasSynced(kluster *v1.Kluster) bool {
+	key, err := cache.MetaNamespaceKeyFunc(kluster)
+	if err != nil {
+		return false
+	}
+	informer, ok := wr.nodeInformerMap.Load(key)
+	if ok {
+		return false
+	}
+	return informer.(NodeInformer).HasSynced()
+}
+
+// AddEventHandlerAddFunc ...
+func (wr *WalleController) addEventHandlerAddFunc(addFunc func(node *api_v1.Node)) {
+	if addFunc != nil {
+		wr.eventHandlerFuncs.Store(funcType.AddFunc, addFunc)
+	}
+}
+
+// AddEventHandlerUpdateFunc ...
+func (wr *WalleController) addEventHandlerUpdateFunc(updateFunc func(curNode, oldNode *api_v1.Node)) {
+	if updateFunc != nil {
+		wr.eventHandlerFuncs.Store(funcType.UpdateFunc, updateFunc)
+	}
+}
+
+// AddEventHandlerUpdateFunc ...
+func (wr *WalleController) addEventHandlerDeleteFunc(deleteFunc func(node *api_v1.Node)) {
+	if deleteFunc != nil {
+		wr.eventHandlerFuncs.Store(funcType.DeleteFunc, deleteFunc)
+	}
+}
+
+// AddEventHandlerFuncs ...
+func (wr *WalleController) AddEventHandlerFuncs(addFunc func(node *api_v1.Node), updateFunc func(curNode, oldNode *api_v1.Node), deleteFunc func(node *api_v1.Node)) {
```

Comment:
the function signatures that are passed are missing  a reference to the Kluster. I think the function need to have a signature like this:
```
addFunc func(kluster *v1.Kluster, node *api_v1.Node), updateFunc func(kluster *v1.Kluster, curNode, oldNode *api_v1.Node), deleteFunc func(kluster *v1.Kluster, node *api_v1.Node)